src/components/file-dropper/file-dropper.tsx

```diff
@@ -0,0 +1,164 @@
+import { Illustration, IllustrationSize } from '../illustration/illustration';
+import { IllustrationType } from '../illustration/illustration-set';
+import { Trans } from 'gatsby-plugin-react-i18next';
+import { TextStyles } from '../typography/typography-v2';
+import { FileListItem } from './file-list-item';
+
+interface FileDropperType {
+  file: File;
+  fileCategory: string | null;
+}
+
+interface FileDropperProps {
+  /**
+   * Called when files are uploaded and all conditions (e.g. `acceptedFileTypes` or `maxFiles`) are fulfilled or when a file changes (e.g. files gets removed or a fileCategory is added to a file). `acceptedFiles` always contains all valid files.
+   */
+  onFileChange: (acceptedFiles: FileDropperType[]) => any;
+  /**
+   * (Optional) Called when files are uploaded and a condition (e.g. wrong file type) is not fulfilled.
+   */
+  onDropRejected?: (rejectedData: FileRejection[]) => any;
+  /**
+   * (Optional) Allows custom validation of uploaded files. If the file is accepted, the `validator()` function must return null. Otherwise the function returns a FileError object. See also: [react-dropzone: custom validation](https://react-dropzone.js.org/#section-custom-validation)
+   */
+  validator?: (file: File) => FileError | FileError[] | null;
+  /**
+   * (Optional) comma-separated list of unique content type specifiers (e.g. `'image/jpeg, image/png, .pdf'`) or `undefined` to allow all files. See also: [react-dropzone: accepting specific file types](https://react-dropzone.js.org/#section-accepting-specific-file-types)
+   */
+  acceptedFileTypes?: string;
+  /**
+   * (Optional) Set a maximum number of files that can be uploaded. If this number is exceeded, `onDropRejected()` is called with all uploaded files.
+   */
+  maxFiles?: number;
+  /**
+   * (Optional) Categories that the user can add to each uploaded file (e.g. 'CV' or 'Cover Letter')
+   * */
+  fileCategories?: string[];
+  /**
+   * (Optional) Pass in a valid illustration keyword to show the according illustration.
+   */
+  illustration?: IllustrationType;
+}
+
+const FileDropperContainer = styled.div<{
+  isDragActive: boolean;
+  hasFiles: boolean;
+}>`
+  display: flex;
+  gap: 16px;
+  flex-direction: column;
+  align-items: center;
+  padding: 28px 0;
+  background-color: ${theme.palette.background.leadbox};
+  cursor: pointer;
+  border: 3px solid
+    ${(props) =>
+      props.isDragActive
+        ? theme.palette.text.link.default
+        : theme.palette.background.leadbox};
+
+  ${(props) =>
+    props.hasFiles &&
+    css`
+      align-items: flex-start;
+      padding: 12px 16px;
+    `}
+
+  &:hover {
+    span {
+      border-bottom: 1px solid ${theme.palette.text.link.default};
+    }
+  }
+`;
+
+const Description = styled.p`
+  ${TextStyles.timestamp};
+  letter-spacing: 0;
+  margin: 0;
+`;
+
+const Highlight = styled.span`
+  color: ${theme.palette.text.link.default};
+`;
+
+export const FileDropper = ({
```

Comment:
I wonder how we are going to use it later. Will it work? Did you spend enough thoughts on that matter or will we discover issues when we integrate it in a form? That could be one nice example in the stories that you can offer after the PR in terms of "how do I actually use it with a form bases on `react-hook-form` (our default) "